components/ws-manager/pkg/manager/manager.go

```diff
@@ -200,6 +200,9 @@ func (m *Manager) StartWorkspace(ctx context.Context, req *api.StartWorkspaceReq
 	}
 	span.LogKV("event", "pod description created")
 
+	// add an annotation to the pod to signal that ws-manager will now attempt to create this pod
+	pod.Annotations[attemptingToCreatePodAnnotation] = "true"
```

Comment:
this should be part of the `createDefiniteWorkspacePod` and corresponding tests (`cd pkg/manager && go test -update -force .`)

Reply:
I disagree. :sweat_smile: 
Adding it into `createDefiniteWorkspacePod` will lose some context as to why it is added, and also this whole retry logic is a temporary workaround for OutOfMemory error. Once we have 1.23.6 k8s and confirm it is working, we might consider removing retry logic all together.
Another reason not to add this into `createDefiniteWorkspacePod` is that it is not obvious that someone has to remove that annotation when pod is created. So if that function will be used later somewhere else, it will have unexpected side effects for someone who is using it.

Reply:
I understand the intent to keep the code together - it is however a break in current style.

Today, all pod creation is covered by the very same createDefiniteWorkspacePod (CDWP) function and tests. I.e. no modifications happen outside of the initial pod struct production prior to creating it in Kubernetes. Adding this annotation outside of the regular pod creation path means we have no coverage that it's added, and we can no longer be sure that what our fixtures say will be used is what's actually used.

`createDefiniteWorkspacePod` only makes sense in the path of creating a pod, and is very unlikely to be used elsewhere.

Also, I don't think this workaround will go away any time soon. When `1.23.6` comes around, we might be able to upgrade in SaaS. But self-hosted users might face the very same problem still. IMHO this code is here to stay for a good while longer.

I have no hard feelings on this (would be ok either way), but a clear preference towards making this part of CDWP for the reasons outlined above.

Reply:
@csweichel I see your point. Fair enough, code and tests updated.